src/coffea/dataset_tools/dataset_query.py

```diff
@@ -528,6 +528,12 @@ def do_preprocess(
         step_size=None,
         align_to_clusters=None,
         scheduler_url=None,
+        recalculate_steps=False,
+        files_per_batch=1,
+        file_exceptions=(OSError,),
+        save_form=False,
+        uproot_options={},
```

Comment:
I think `None` is best here due to the known problems of mutable default args in python. Other than that I don't see a problem here.

Reply:
In general mutable default options are bad, but as this is in a library where we don't modify it between this point and uproot.open, maybe it can stay (and this is a duplicate of coffea's default in preprocess). @lgray should that be changed to None -> {}? 
https://github.com/CoffeaTeam/coffea/blob/master/src/coffea/dataset_tools/preprocess.py#L68
Trying to unpack `None` would give an error here, so we can't permit passing through a `None` option

Reply:
Ah, I didn't notice what is already there in the coffea code. I always do `None` and then `if x is None: x={}` in my code so that no linters complain.

Reply:
`**kwargs` will be a `dict` even if no kwargs are passed.

from the python 3.11 repl:
```
>>> def fun(**kwargs):
...     print(type(kwargs))
...
>>> fun()
<class 'dict'>
```

and preprocess also declares the type as `dict` with no nullability.

Reply:
hence, it should always be a dict.